File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        ans=0
        freq={}
        for num in nums:
            freq[num]=freq.get(num,0)+1


        for num in freq:
            if num+1 in freq:
                ans=max(ans,freq[num]+freq[num+1])

        return ans                

File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.java
class Solution {
    public int findLHS(int[] nums) {
        HashMap<Integer,Integer> freq= new HashMap<>();
        int ans=0;
        for (int num : nums){
            freq.put(num,freq.getOrDefault(num,0)+1);
            }

        for (int num:freq.keySet()){
            if(freq.containsKey(num+1)){
                ans=Math.max(ans,freq.get(num)+freq.get(num+1));
            }

        }  
        return ans;

        }
        
    }
